sympy/matrices/common.py

```diff
@@ -753,6 +755,40 @@ def size(m):
                         diag_entries[(i + row_pos, j + col_pos)] = m[i, j]
                 row_pos += m.rows
                 col_pos += m.cols
+            elif(isinstance(m,dict)):
+                # in this case we're a dict
+                for key,value in m.items():
+                    r_p = 0
```

Comment:
since you redefine these in the next two lines, these two lines are not necessary